src/parser.ts

```diff
@@ -187,9 +183,197 @@ function getFunctionName(func: ts.FunctionLikeDeclaration): string {
+    } else if (ts.isVariableDeclaration(parent)) {
+        if (ts.isIdentifier(parent.name)) {
+            return getPropertyName(parent.name);
+        }
+    } else if (
+        ts.isBinaryExpression(parent) &&
+        parent.operatorToken.kind === ts.SyntaxKind.EqualsToken
+    ) {
+        if (ts.isIdentifier(parent.left)) {
+            return getPropertyName(parent.left);
+        } else if (
+            ts.isPropertyAccessExpression(parent.left) ||
+            ts.isElementAccessExpression(parent.left)
+        ) {
+            return getLeftHandSideName(parent.left);
+        }
+    }
+    return prefixWithObject ? "<Object>" : null;
+}
+
+function getParentClassProperty(func: ts.FunctionLikeDeclaration) {
+    let { member } = getMemberNodeForFunction(func);
+    if (member) {
+        const parent = member.parent;
+        if (ts.isClassDeclaration(parent) || ts.isClassExpression(parent)) {
+            return parent;
+        }
+    }
+    return null;
+}
+
+function getClassPrefix(func: ts.FunctionLikeDeclaration): string | null {
+    let { member, isProperty } = getMemberNodeForFunction(func);
+    if (member) {
+        const parent = member.parent;
+        if (ts.isClassDeclaration(parent) || ts.isClassExpression(parent)) {
+            const className = getClassName(parent);
+            if (isStatic(member)) {
+                if (!isProperty || func.name === undefined) {
+                    return className;
+                }
+            } else if (ts.isConstructorDeclaration(func)) {
+                return className;
+            } else if (!isProperty) {
+                return `${className}.prototype`;
+            }
+        }
+    }
+    return null;
+}
+
+function getClassName(classNode: ts.ClassLikeDeclaration): string {
+    // if class has a local name, use it and return,
+    // not interested in left hand side
+    if (classNode.name) {
+        return classNode.name.text;
+    }
+    // try to get the name from the left hand side
+    let nameText = null;
+    if (ts.isClassExpression(classNode)) {
+        const parent = classNode.parent;
+        if (
+            ts.isVariableDeclaration(parent) ||
+            ts.isPropertyAssignment(parent) ||
+            ts.isPropertyDeclaration(parent)
+        ) {
+            if (ts.isPropertyName(parent.name)) {
+                nameText = getPropertyName(parent.name);
+            }
+        } else if (
+            ts.isBinaryExpression(parent) &&
+            parent.operatorToken.kind === ts.SyntaxKind.EqualsToken
+        ) {
+            if (
+                ts.isPropertyAccessExpression(parent.left) ||
+                ts.isElementAccessExpression(parent.left)
+            ) {
+                nameText = getLeftHandSideName(parent.left);
+            }
+        }
+    }
+    return nameText ? nameText : "<anonymous>";
+}
+
+function isStatic(func: ts.Node): boolean {
+    if (func.modifiers) {
```

Comment:
A destructured arrow function would simplify the some argument.

Reply:
Is it possible for a function to not have a modifiers property?

Reply:
It is. The type of `modifiers` is `modifiers?: ModifiersArray;`